sound/soc/sof/ipc4-topology.c

```diff
@@ -1501,8 +1463,29 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 		return -EINVAL;
 	}
 
-	/* set the gateway dma_buffer_size using the matched ID returned above */
-	copier_data->gtw_cfg.dma_buffer_size = available_fmt->dma_buffer_size[ret];
+	/*
+	 * set the gateway dma_buffer_size to twice the size of the input/output buffer size
+	 * by default to meet the FW expectation. In the deep buffer case, set the dma_buffer_size
+	 * depending on the deep_buffer_dma_ms set in topology.
+	 */
+	switch (swidget->id) {
+	case snd_soc_dapm_dai_in:
+		copier_data->gtw_cfg.dma_buffer_size = 2 * copier_data->base_config.ibs;
+		break;
+	case snd_soc_dapm_aif_in:
+		if (deep_buffer_dma_ms)
+			copier_data->gtw_cfg.dma_buffer_size =
+				deep_buffer_dma_ms * copier_data->base_config.ibs;
```

Comment:
humm, what if the token value is 1ms only, would that work? wondering is this should min(2, deep_buffer_dma_ms)

Reply:
makes sense. fixed now